src/mantine-nprogress/src/NProgressProvider.tsx

```diff
@@ -0,0 +1,102 @@
+import { MantineTransition, Progress, Transition } from '@mantine/core';
+import { useDebouncedValue, useDidUpdate, useInterval } from '@mantine/hooks';
+import { MantineColor } from '@mantine/styles';
+import React, { useState } from 'react';
+import { nprogressContext } from './NProgress.context';
+
+export interface NProgressProviderProps {
+  /** Your app */
+  children: React.ReactNode;
+  progress?: number;
+  color?: MantineColor;
+  size?: number;
+  onFinish?: () => void;
+  /** Transition duration in ms */
+  transitionDuration?: number;
+  /** Predefined transition name */
+  transition?: string;
+  /** Interval time in ms */
+  intervalTime?: number;
+  /** Interval step size */
+  intervalStep?: number;
+  /** Unmounted time after progress is 100% */
+  unmountTime?: number;
+  /** Unmount duration in ms */
+  unmountDuration?: number;
+  /** Unmount transition function */
+  unmountTransition?: MantineTransition;
+}
+
+export function NProgressProvider({
+  children,
+  color = 'blue',
+  progress = 0,
+  size = 2,
+  intervalTime = 500,
+  intervalStep = 1,
+  transition = 'ease',
+  transitionDuration = 600,
+  unmountTime = intervalTime + 200,
+  unmountDuration: exitDuration = 600,
+  unmountTransition = 'fade',
+  onFinish,
+}: NProgressProviderProps) {
+  const [_progress, setProgress] = useState(progress);
+  const [mounted] = useDebouncedValue(_progress !== 100, unmountTime);
+
+  const interval = useInterval(
```

Comment:
I do not see much purpose in interval, it should be removed.

Reply:
The interval is for the start and stop function so I can increment the progress bar